pkg/servicemanager/startuprecorder/startuprecorder.go

```diff
@@ -0,0 +1,71 @@
+package startuprecorder
+
+import (
+	"encoding/json"
+	"os"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+type ServiceData struct {
+	Name         string        `json:"name"`
+	Dependencies []string      `json:"dependencies"`
+	Start        time.Time     `json:"start"`
+	Ready        time.Time     `json:"ready"`
+	TimeToReady  time.Duration `json:"timeToReady"`
+}
+
+type StartupRecorder struct {
+	Services              []ServiceData `json:"services"`
+	MicroshiftStart       time.Time     `json:"microshiftStart"`
+	MicroshiftReady       time.Time     `json:"microshiftReady"`
+	MicroshiftTimeToReady time.Duration `json:"microshiftTimeToReady"`
+
+	m sync.Mutex
+}
+
+func New() *StartupRecorder {
+	return &StartupRecorder{}
+}
+
+func (l *StartupRecorder) LogServiceReady(serviceName string, dependencies []string, start time.Time, ready time.Time) {
+	serviceData := ServiceData{
+		Name:         serviceName,
+		Dependencies: dependencies,
+		Start:        start,
+		Ready:        ready,
+		TimeToReady:  ready.Sub(start),
+	}
+
+	l.m.Lock()
+	l.Services = append(l.Services, serviceData)
+	l.m.Unlock()
+
+	klog.InfoS("SERVICE READY", "service", serviceName, "since-start", ready.Sub(start))
+}
+
+func (l *StartupRecorder) LogMicroshiftStart(start time.Time) {
+	klog.InfoS("MICROSHIFT STARTING")
+	l.MicroshiftStart = start
+}
+
+func (l *StartupRecorder) LogMicroshiftReady(ready time.Time) {
+	klog.InfoS("MICROSHIFT READY", "since-start", time.Since(l.MicroshiftStart))
+	l.MicroshiftReady = ready
+	l.MicroshiftTimeToReady = ready.Sub(l.MicroshiftStart)
+}
+
+func (l *StartupRecorder) OutputData() error {
+	jsonOutput, _ := json.Marshal(l)
```

Comment:
You should not ignore the error. We should at least capture it and `log.Error`. Otherwise, in next lines, we'll get empty log and empty file.

This function's signature suggests it returns an error, but it's really not.
Obviously we don't want to ruin the startup of microshift by failing to serialize and write the data, so we should remove the `error` as a returned type.